Cubical/Algebra/Kernel.agda

```diff
@@ -0,0 +1,48 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+module Cubical.Algebra.Kernel where
```

Comment:
Calling this file Kernel is maybe not ideal (pun intended!) as it's just about kernels in rings... Maybe we should have a subfolder Algebra.Ring where we have definitions like this? Alternatively calling thing `RingKernel` or something, but I prefer adding a subfolder

Reply:
I didn't think about it, I like the subfolder idea.

Reply:
I'll also move Ideal and QuotientRing to Ring/